pkg/maps/nat/nat.go

```diff
@@ -84,10 +85,15 @@ func NewMap(name string, v4 bool, entries int) *Map {
 		mapValue = &NatEntry6{}
 		sizeVal = int(unsafe.Sizeof(NatEntry6{}))
 	}
+	if lru {
+		mapType = bpf.MapTypeLRUHash
+	} else {
+		mapType = bpf.MapTypeHash
+	}
 	return &Map{
 		Map: *bpf.NewMap(
 			name,
-			bpf.MapTypeLRUHash,
+			mapType,
```

Comment:
cc @joestringer

Reply:
What's the goal of forcing the caller to know about the map type being LRU?

Each map in Cilium has a preferred type and then we typically support fallback. So for instance in the case of NAT map, we define here that it should ideally be LRU. When the `map.Open()` is called, `pkg/bpf/map.go:GetMapType()` will resolve whether the LRU type is supported and potentially fall back.

Elsewhere in this patch, you're doing a new `Create()` function which calls `openOrCreate()` which also calls `GetMapType()` to resolve the actual type that will be used.

To answer the upgrade question, things like LRU support are determined by the underlying kernel version. While *maybe* in theory there's an upgrade path there where someone swaps the kernel from under us, I don't know of any common cases where this will occur; it's far more likely that the node is restarted and Cilium is therefore restarted as well. At that point I don't see any upgrade issues.

Reply:
> When the `map.Open()` is called, `pkg/bpf/map.go:GetMapType()` will resolve whether the LRU type is supported and potentially fall back.

Ah, I didn't know that. I can remove this commit then and it solves both this concern and the other below :tada: